docs/GlobalHelperFunctionsFromExternalDrivers.md

```diff
@@ -0,0 +1,96 @@
+# Global Helper Functions from External Drivers
+
+## Overview
+
+This document outlines the requirements for enabling drivers outside of `ebpfcore.sys` to expose global helper functions to eBPF programs without requiring centralized management of helper function IDs.
+
+## Problem Statement
+
+The current eBPF for Windows global helper function system requires:
+- **Global helpers hosted only in `ebpfcore.sys`** limiting extensibility by third-party drivers
+- **Pre-assigned numeric IDs** for all helper functions
+- **Centralized coordination** of helper ID assignments to prevent conflicts
+- **Static ID management** making it difficult for external drivers to expose custom global helper functions
+
+This creates barriers for independent driver development and limits the extensibility of the eBPF ecosystem by preventing external drivers from contributing global helper functions.
+
+## Requirements
+
+### R1: External Driver Global Helper Registration
+- Drivers outside of `ebpfcore.sys` **MUST** be able to register global helper functions as general program information providers
+- The system **MUST** support global helper function registration using descriptive names instead of pre-assigned numeric IDs
+- Registration **MUST** use existing NMR (Network Module Registrar) interfaces without modification
+- External drivers **MUST** be able to register global helpers with `helper_id = -1` to indicate name-based registration
```

Comment:
This is an example of a requirement that is phrased as requiring a particular solution.  Using BTF IDs should meet the requirements.  (Unless you add new requirements and explain why BTF IDs fail.)

Please add a requirement along the lines of "Consistent with BPF ISA standard".  Helper IDs are debatable since they're considered "historical" not recommended.  BTF IDs are certainly consistent with the BPF ISA standard.